Objects/obmalloc.c

```diff
@@ -1580,42 +1580,43 @@ allocate_from_new_pool(uint size)
 
 /* pymalloc allocator
 
-   Return 1 if pymalloc allocated memory and wrote the pointer into *ptr_p.
+   Return a pointer to newly allocated memory if pymalloc allocated memory.
 
-   Return 0 if pymalloc failed to allocate the memory block: on bigger
+   Return NULL if pymalloc failed to allocate the memory block: on bigger
    requests, on error in the code below (as a last chance to serve the request)
    or when the max memory limit has been reached.
 */
-static inline int
-pymalloc_alloc(void *ctx, void **ptr_p, size_t nbytes)
+static inline void*
+pymalloc_alloc(void *ctx, size_t nbytes)
 {
 #ifdef WITH_VALGRIND
     if (UNLIKELY(running_on_valgrind == -1)) {
         running_on_valgrind = RUNNING_ON_VALGRIND;
     }
     if (UNLIKELY(running_on_valgrind)) {
-        return 0;
+        return NULL;
     }
 #endif
 
     if (UNLIKELY(nbytes == 0)) {
-        return 0;
+        return NULL;
     }
     if (UNLIKELY(nbytes > SMALL_REQUEST_THRESHOLD)) {
-        return 0;
+        return NULL;
     }
 
     uint size = (uint)(nbytes - 1) >> ALIGNMENT_SHIFT;
     poolp pool = usedpools[size + size];
     block *bp;
-    
+
     if (LIKELY(pool != pool->nextpool)) {
         /*
          * There is a used pool for this size class.
          * Pick up the head block of its free list.
          */
         ++pool->ref.count;
         bp = pool->freeblock;
+        assert(bp != NULL);
 
         if (UNLIKELY((pool->freeblock = *(block **)bp) == NULL)) {
             // Reached the end of the free list, try to extend it.
```

Comment:
It's better to use comment style "/* */" instead of "//" here，to ensure consistent context.
```suggestion
            /* Reached the end of the free list, try to extend it.
             */
```

Reply:
This code was likely written before PEP 7 was updated to allow a subset of C99. My PR doesn't add these comments ;-)